script/github-actions/create-cross-app-import-graph.js

```diff
@@ -0,0 +1,296 @@
+/* eslint-disable no-param-reassign */
+const fs = require('fs');
+const path = require('path');
+const findImports = require('find-imports');
+const core = require('@actions/core');
+
+function getImports(filePath) {
+  return findImports(filePath, {
+    absoluteImports: true,
+    relativeImports: true,
+    packageImports: false,
+  });
+}
+
+function diffIncludesSrcApplicationsFiles(diff) {
+  return diff.includes('diff --git a/src/applications');
+}
+
+function getAppNameFromFilePath(filePath) {
+  return filePath.split('/')[2];
+}
+
+function getAppPathFromFileDiff(fileDiff) {
+  const str = fileDiff.replace('diff --git a/', '');
+  return str.slice(0, str.indexOf(' '));
+}
+
+/* Function iterates through the characters in a diff and
+ * and extracts, to an array, each part that pertains to
+ * changes to a src/application file, then returns the array
+ */
+function sliceDiffIntoDiffForEachChangedFile(diff) {
+  const diffForEachChangedFile = [];
+  const startOfChange = 'diff --git a/src/applications';
+  const endOfChange = 'diff --git a/';
+  let startOfDiffIndex = null;
+
+  for (let i = 0; i <= diff.length; i += 1) {
+    if (
+      !startOfDiffIndex &&
+      diff[i] === startOfChange[0] &&
+      diff.slice(i, i + startOfChange.length) === startOfChange
+    ) {
+      startOfDiffIndex = i;
+    } else if (
+      startOfDiffIndex &&
+      endOfChange &&
+      diff[i] === endOfChange[0] &&
+      diff.slice(i, i + endOfChange.length) === endOfChange
+    ) {
+      diffForEachChangedFile.push(diff.slice(startOfDiffIndex, i));
+      i -= 1; // reduce i by 1 so the next iteration picks up the beginning of this diff
+      startOfDiffIndex = null;
+    } else if (startOfDiffIndex && i === diff.length - 1) {
+      diffForEachChangedFile.push(diff.slice(startOfDiffIndex));
+    }
+  }
+
+  return diffForEachChangedFile;
+}
+
+function isSrcAppicationFileDiff(fileDiff) {
+  return fileDiff.replace('diff --git a/', '').startsWith('src/applications');
+}
+
+function getSrcApplicationDiffs(diff) {
+  return sliceDiffIntoDiffForEachChangedFile(diff).filter(fileDiff => {
+    return isSrcAppicationFileDiff(fileDiff);
+  });
+}
+
+/* Function takes an import reference and returns the path
+ * to the referenced file from 'src/' if the reference begins
+ * with 'applications/' or starts with '../'. Otherwise
+ * it returns the given reference.
+ */
+function getImportPath(filePathAsArray, importRef) {
+  if (importRef.startsWith('applications/')) {
+    return `src/${importRef}`;
+  } else if (importRef.startsWith('../')) {
+    const numDirsUp = importRef.split('/').filter(str => str === '..').length;
+
+    return importRef.replace(
+      '../'.repeat(numDirsUp),
+      `${filePathAsArray
+        .slice(0, filePathAsArray.length - 1 - numDirsUp)
+        .join('/')}/`,
+    );
+  }
+
+  return importRef;
+}
+
+function importIsFromOtherApplication(appName, importPath) {
```

Comment:
Do we need to consider imports from `/src/platform`?

Reply:
No, tests in `src/platform` always run by default.